Updates for nuget package type

eeclfrei opened the pull request.

DanielJoyce committed:
Add type alias for UpdateProjectRequest

DanielJoyce committed:
Add a type alias for the create project response for update responses

louislang committed:
Merge pull request #4 from phylum-dev/daniel/add-type-alias

Daniel/add type alias

DanielJoyce committed:
Add untagged variant to wrap Job status response variants so they can be expressed as return types for a single endpoint

ejortega committed:
Add updates for maven

eeclfrei committed:
Add .NET ecosystem type

ejortega committed:
Add ref for maven projects

eeclfrei committed:
Standardize package type names for rubygems / pypi

ejortega committed:
Add improvements for accessing useful struct members (#14)

DanielJoyce committed:
Daniel/fix project types (#19)

* In support of ongoing work on the api side, add a feature
flag to gate certain changes that are being worked on

* Update project summary response to use timestamps
* Add created_at field
* Added feature flag for ongoing migration work on api side

* Revert version field to 0.1.0 as api doesn't change unless feature gate enabled

mathew-horner committed:
Implement Display interface for RefreshToken so we can print it out with phylum auth token.

kylewillmon commented:
The `master` and `development` branches are in a pretty weird state. Consider force pushing `master` when this merge happens

cd-work committed:
Add derives for common traits (#21)

To ensure that the `phylum-types` crate doesn't have to be constantly
changed to allow for dependents to use specific traits for the types
inside this crate, all common traits have been implemented for all
existing types.

The chosen traits are based on
https://rust-lang.github.io/api-guidelines/interoperability.html and
have been derived wherever possible without worrying too much about
their intended usecase. This should especially be helpful for test code
which is usually a bit more practical than efficient.

Allan-Clements committed:
Group Accounts API structs (#23)

* Added Group Accounts API response structs, made chrono dependency no longer optional, also added optional group_name for create project request

* Update src/types/group.rs

Co-authored-by: Christian Dürr <102963075+cd-work@users.noreply.github.com>

* Use bool default of false instead of None for is_admin and is_owner given both would mean the same thing

Co-authored-by: Christian Dürr <102963075+cd-work@users.noreply.github.com>

Allan-Clements committed:
Restore dev_api_issue_96 feature flag (#24)

cd-work committed:
Add Display impl for all token types (#26)

cd-work committed:
Allow freely setting project thresholds (#25)

Previously the `active` field of the thresholds struct was populated by
checking whether the `threshold` was above `0` or not. However the
`Total Project Threshold` can never be disabled and it is possible for
other thresholds to be at `0` without being disabled too.

To ensure a more accurate representation of the backend state, the
`set_threshold` method instead takes a `Threshold` instance which can be
populated fully by the consumer without any internal restrictions.

See #288.

cd-work committed:
Fix code formatting

cd-work committed:
Add GitHub Actions CI

cd-work committed:
Skip serialization of `group_name` when `None` (#30)

See phylum-dev/cli#331.

Allan-Clements committed:
Removed AddUserToGroupRequest since we url encode the target instead (#31)

Allan-Clements committed:
Add group_name to SubmitPackageRequest (#32)

cd-work committed:
Add new package endpoint types (#33)

cd-work committed:
Fix package types for API integration (#34)

These are some minor changes necessary to the new package endpoint types
introduced in ccd3449, which are necessary for integration into
phylum-dev/api.

kylewillmon committed:
Accept risk_level and risk_domain when deserializing Issues (#35)

mathew-horner committed:
Add JSON Schema to types (#36)

louislang committed:
Automatically add issues to main project board (#37)

cd-work committed:
Fix pom otherArchives data format

See phylum-dev/cli#440.

samtay committed:
Add group name to projects

samtay committed:
Remove dev_api_issue_96

mathew-horner committed:
Add ecosystem field to ProjectSummaryResponse (#39)
